precheck.sh

```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-sbt clean coverage test it:test scalafmt::test test:scalafmt::test coverageReport
+sbt clean coverage test it:test scalafmt::test test:scalafmt::test scalastyle coverageReport
```

Comment:
:thinking: Probably we could move `scalafmt::test test:scalafmt::test` as the first step, so that it fails fast.

Reply:
Perhaps, but if the tests fail and you have to change your code and you want the code to function correctly before you care about its style! :) Number of times my pipeline failed because of some silly scalastyle contravention I put this here so I wouldn't forget in future! :D